src/communication_mode.c

```diff
@@ -0,0 +1,29 @@
+// Copyright 2025 Shift Cryptosecurity AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "communication_mode.h"
+
+#include "memory/memory_shared.h"
+
+static bool _usb_hww_request_seen = false;
+
+void communication_mode_enable_usb(void)
+{
+    _usb_hww_request_seen = true;
+}
+
+bool communication_mode_is_ble(void)
+{
+    return !_usb_hww_request_seen && memory_get_platform() == MEMORY_PLATFORM_BITBOX02_PLUS;
```

Comment:
this is quite a lot less efficient since it always goes out and reads flash in case `_usb_hww_request_seen==false`.

You could add a 
```
static has_ble = memory_get_platform() == MEMORY_PLATFORM_BITBOX02_PLUS;
return !_usb_hww_request_seen && has_ble;
```